### PR TITLE
feat(dev): Use arm64 ClickHouse image on Apple arm64 machines

### LINKED ISSUE #28564: No clickhouse Docker image for Mac Arm64
Request for an official image: https://github.com/ClickHouse/ClickHouse/issues/22222

Building Clickhouse is also broken for Apple M1 machines:
https://github.com/ClickHouse/ClickHouse/issues/27593

### PR DESCRIPTION
**Pull request by armenzg**
ClickHouse, currently, [does not provide an `arm64` Docker image for the clickhouse-server][issue].

Until they do, we can use the [experimental image that Altinity produces][image].
I believe this image is built by this [script][script] on an `aarch64` host.

Fixes #28564

[issue]: https://github.com/ClickHouse/ClickHouse/issues/22222
[image]: https://hub.docker.com/layers/altinity/clickhouse-server/21.6.1.6734-testing-arm/images/sha256-9a4516444fef9e0f11ee6b2de716d3b97b36bf05d9cc2d44c4596cfb0584dea6?context=explore
[script]: https://gist.github.com/filimonov/5f9732909ff66d5d0a65b8283382590d

### CODE REVIEW

File: config/clickhouse/loc_config.xml [hunk 1]
```diff
@@ -1,3 +1,3 @@
 <yandex>
-    <max_server_memory_usage_to_ram_ratio from_env="MAX_MEMORY_USAGE_RATIO" />
+    <max_server_memory_usage_to_ram_ratio>0.3</max_server_memory_usage_to_ram_ratio>
```
Comment (armenzg) [inline config/clickhouse/loc_config.xml:2]: Unfortunately, the arm64 image does *not* read the value from memory.

**Comment by armenzg** [comment]
On an Apple M1 machine this can be tested with `sentry devservices up snuba clickhouse`.

File: config/clickhouse/loc_config.xml (hunk 1 above)
Comment (lynnagara) [reply config/clickhouse/loc_config.xml:2]: Please also update this in https://github.com/getsentry/sentry/blob/master/config/clickhouse/dist_config.xml#L2

File: src/sentry/conf/server.py [hunk 2]
```diff
@@ -1780,15 +1783,22 @@ def build_cdc_postgres_init_db_volume(settings):
     ),
     "clickhouse": lambda settings, options: (
         {
-            "image": "yandex/clickhouse-server:20.3.9.70",
+            # altinity provides clickhouse support to other companies
+            # Official support: https://github.com/ClickHouse/ClickHouse/issues/22222
+            # This image is build with this script https://gist.github.com/filimonov/5f9732909ff66d5d0a65b8283382590d
+            "image": "altinity/clickhouse-server:21.6.1.6734-testing-arm"
```
Comment (lynnagara) [inline src/sentry/conf/server.py:1789]: Is this the earliest ClickHouse version that works with ARM? That's a big jump from 20.3 to 21.6.

**Review by BYK — CHANGES_REQUESTED** [review]
Mostly rejecting for the version mismatch.

**Review by BYK — APPROVED** [review]

**Pull request merged**